design-proposals/amd-sev.md

```diff
@@ -0,0 +1,253 @@
+
+- VM gets scheduled on a SEV-capable node
+- guest owner validates the platform on that specific node
+- guest owner establishes a secure channel with the platform by providing the session parameters
+- VM is started in a paused state
+- guest owner requests and verifies the measurement
+- guest owner signals the VM to unpause by injecting the secret
+
+More details about each individual step are provided bellow.
+
+### Step #1 Validation of the platform
+
+The guest owner should get the assets listed bellow from the platform owner:
+
+- the `PDH` (Platform Diffie-Hellman) key exported from the SEV hardware
+- the complete key chain (up to the `root of trust`) used to sign the `PDH`
+- the OVMF binary
+
+KubeVirt can expose a VMI subresource endpoint `/sevfetchcertchain` which will return base64 encoded certificates. Also a new command can be added to virtctl tool to facilitate the process: `virtctl fetchcertchain ...`.
+
+On a node where KubeVirt is running, the privileged `virt-handler` pod can export the required assets in advance to a `kubevirt-private` directory. Whenever a new VM requiring SEV is launched, the directory can be mounted to the `virt-launcher` pod in `read-only` mode. A user can then invoke the `virtctl` tool to fetch the certificate chain for validation.
+
+In order to ensure that the platform is not compromised the guest owner needs to validate the `PDH` key using the provided chain. For that `sev-tool --validate_cert_chain ...` can be used (AMD root certificate should be downloaded from the web for verification). Additionally the UEFI image should be considered 'trustable' by the guest owner.
+
+### Step #2 Establishing secure channel with the platform
+
+To establish a secure communication channel with the platform a guest owner needs to provide the launch blob with the parameters of the session and the certificate derived from `PDH`. This can be suported in KubeVirt by `/sevsetupsession` VMI endpoint with the parameters `session` and `dhCert` (which will be mapped to the corresponding elements in the domain xml). New `virtctl sevsetupsession ...` command can be introduced to trigger propagation of the parameters.
+
+The implementation in `virt-launcher` in turn can 'postpone' the creation of the domain until the parameters are specified. Sidecar hooks functionality can be used to modify the domxml after a VMI has been scheduled.
```

Comment:
I am not sure if the sidecar sentence has any relevance here?

Reply:
Removed for now.